### PR TITLE
Feat : Added Watchlater

### PR DESCRIPTION
**Pull request by srejitk**
The following merge request adds the following ;

1. VideoContext
2. UserContext
3. VideoCard
4. HorizontalVideoCard
5. Watchlater Page
6. Protected Routes

### CODE REVIEW

File: src/context/Video/VideoContext.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { createContext, useContext, useReducer, useEffect } from "react";
+import useAxios from "../../utils/useAxios";
+import { VideoReducer } from "./VideoReducer";
+
+VideoReducer;
```
Comment (rahulcs754) [inline src/context/Video/VideoContext.jsx:5]: i think you should remove this

File: src/context/Video/Watchlater.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        authorization: localStorage.getItem("Token"),
+      },
+    });
+    const { status, data } = response;
+    if (status === 200) {
+      videoDispatch({
+        type: "DELETE_FROM_WATCHLATER",
+        payload: data?.watchlater,
+      });
+      Toast({ message: "Deleted from Watchlist", type: "success" });
```
Comment (rahulcs754) [inline src/context/Video/Watchlater.jsx:57]: nice to create api function to seperate code that will look good

File: src/pages/Explore/Explore.jsx [hunk 3]
```diff
@@ -1,10 +1,12 @@
-import React from "react";
+import React, { useState } from "react";
 import styles from "./Explore.module.css";
 import { videos } from "../../backend/db/videos";
```
Comment (rahulcs754) [inline src/pages/Explore/Explore.jsx:3]: i think you should access all video using api that will give you list of video. if you have any doubt. you can talk with me

**Review by rahulcs754 — APPROVED** [review]
The code looks good and clean .
suggest small suggestion

**Review by nakulsharma15 — APPROVED** [review]
Well Structured Code. No changes suggested. I liked how you named the CSS classes and the overall folder structure.

File: src/pages/Explore/Explore.jsx (hunk 3 above)
Comment (srejitk) [reply src/pages/Explore/Explore.jsx:3]: Thank you for bringing this to my notice, deleting that import statement.

**Comment by srejitk** [comment]
Thank You

File: src/context/Video/VideoContext.jsx (hunk 1 above)
Comment (srejitk) [reply src/context/Video/VideoContext.jsx:5]: Corrected. Thank You.

**Pull request merged**